pkg/infra/metrics/metrics.go

```diff
@@ -589,3 +591,25 @@ func newCounterStartingAtZero(opts prometheus.CounterOpts, labelValues ...string
 
 	return counter
 }
+
+// SanitizeLabelName removes all invalid chars from the label name.
+// If the label name is empty or contains only invalid chars, it
+// will return an error.
+func SanitizeLabelName(name string) (string, error) {
+	if len(name) == 0 {
+		return "", errors.New("label name cannot be empty")
+	}
+
+	out := strings.Builder{}
+	for i, b := range name {
+		if (b >= 'a' && b <= 'z') || (b >= 'A' && b <= 'Z') || b == '_' || (b >= '0' && b <= '9' && i > 0) {
+			out.WriteRune(b)
+		}
+	}
+
+	if out.Len() == 0 {
+		return "", errors.New("label name only contains invalid chars")
```

Comment:
IMO, including the erroneous name would help in debugging?
```suggestion
		return "", fmt.Errorf("label name only contains invalid chars: %q", name)
```